### PR TITLE
ISPN-14230 Download server report from cluster membership list

### PR DESCRIPTION
**Pull request by dpanshug**
* Action added to download server report.
* Table changed to TableComposable
* Labels exported

https://issues.redhat.com/browse/ISPN-14230

### CODE REVIEW

File: src/app/ClusterStatus/ClusterStatus.tsx [hunk 1]
```diff
@@ -21,108 +21,65 @@ import {
...
+    name: t('cluster-membership.node-name'),
+    physicalAdd: t('cluster-membership.physical-address')
+  };
+
+  const clusterMemberActions = (row): IAction[] => [
     {
-      title: 'Physical address'
+      title: t('cluster-membership.download-report'),
+      onClick: () => {
+        ConsoleServices.server().downloadReport(row.name);
```
Comment (karesti) [inline src/app/ClusterStatus/ClusterStatus.tsx:52]: can;t this be done with a hook ?
would be nice to try to unit test with jest mocking the server report download calls

File: src/services/serverService.ts [hunk 2]
```diff
@@ -21,4 +21,15 @@ export class ServerService {
   public async getVersion(): Promise<Either<ActionResponse, string>> {
     return this.utils.get(this.endpoint, (data) => data.version);
   }
+
+  /**
+   * Get server report for a given nodeName
+   */
+  public downloadReport(nodeName: string): void {
```
Comment (karesti) [inline src/services/serverService.ts:28]: server errors are not handled there

Comment (karesti) [reply src/services/serverService.ts:28]: is this the way it works in Patternfly ?

File: src/app/ClusterStatus/ClusterStatus.tsx (hunk 1 above)
Comment (dpanshug) [reply src/app/ClusterStatus/ClusterStatus.tsx]: Added hook and unit test.

File: src/services/serverService.ts (hunk 2 above)
Comment (dpanshug) [reply src/services/serverService.ts]: Changed function to handle server error. 
There's no mention of the download feature in patternfly.

**Comment by dpanshug** [comment]
One thing I've noticed, downloaded reports from server `./run-server-for-e2e.sh` are not extractable 
image

Whereas it's working fine with server `15.0.0-SNAPSHOT` 

@andyuk1986 can you check pls.

**Comment by karesti** [comment]
@jabolina any clue with @dpanshug issue ?

**Comment by andyuk1986** [comment]
Let me check it out

**Comment by andyuk1986** [comment]
@dpanshug @karesti seems that when we run the maven command to install the latest console changes to the server, i.e. build the console and copy the dist to the static directory of the already running server, then the generated report is corrupted. But if we will restart the server after that action, then the server is generated normally. Maybe we need to make some changes to script and maven commands so that later when tests are running we get normally generated report?

**Comment by dpanshug** [comment]
> @dpanshug @karesti seems that when we run the maven command to install the latest console changes to the server, i.e. build the console and copy the dist to the static directory of the already running server, then the generated report is corrupted. But if we will restart the server after that action, then the server is generated normally. Maybe we need to make some changes to script and maven commands so that later when tests are running we get normally generated report?

Yes, after restarting server reports are extractable.

But the other thing, after comparing reports from both server `15.0.0-SNAPSHOT` and `./run-server-e2e`, server `./run-server-e2e` lacks some files and folder, as you can see in screenshot below

(left: `./run-server-e2e`, right: `15.0.0-SNAPSHOT`)
Screenshot from 2023-04-12 12-20-28

Is this okay? @jabolina

**Comment by karesti** [comment]
The feature is ok from the console, but we need a release of Infinispan Dev in 15 for testing

**Comment by karesti** [comment]
integration testing needs to be done with a released server

**Comment by jabolina** [comment]
The missing files/folders can happen when some of the commands are not available in the system, or, like in this case, if it is unable to copy some of the folders from the server's root. That can happen because of file ownership, for example. +1 for Katia's suggestion of running on a released server.

**Review by karesti — COMMENTED** [review]
LGTM but we need a server release so will be on hold now

**Comment by karesti** [comment]
@dpanshug @andyuk1986 the server Dev01 has been released so we can change the version in main, 14.0 branch will remain with 14.0 server

**Review by karesti — CHANGES_REQUESTED** [review]
use 15.0.0.Dev01 server here

**Comment by dpanshug** [comment]
Feature is not working with the released 15.0.0.Dev01 :roll_eyes: 
@jabolina can you confirm if the API is released?

**Comment by jabolina** [comment]
I was able to use it on 15.0.0.Dev01 image.

```bash
$ docker pull infinispan/server:15.0.0.Dev01
$ docker run -it -p 11222:11222 -e USER="admin" -e PASS="password" infinispan/server:15.0.0.Dev01
```

On the other terminal:

```bash
$ curl --digest -u admin:password -O 'http://127.0.0.1:11222/rest/v2/server/report/f439e0909dbd-59934'
```

The `f439e0909dbd-59934` was the name my server received. The name is printed on the docker output.

**Review by karesti — CHANGES_REQUESTED** [review]
works well
the only issue I see is that there is no notification when the download has started.
I click the button and there is some time before the download is happening, so I don't know if it worked or not.
I would add an alert to say the download is happening (green) or that has failed for any reason with the error altert.

File: src/app/ClusterStatus/ClusterStatus.tsx [hunk 3]
```diff
@@ -195,18 +143,63 @@ const ClusterStatus: React.FunctionComponent<any> = (props) => {
...
+              ) : (
+                filteredClusterMembers.map((row) => {
+                  return (
+                    <Tr key={row.name}>
+                      <Td dataLabel={columnNames.name}>{row.name}</Td>
+                      <Td dataLabel={columnNames.physicalAdd}>{row.physical_address}</Td>
+                      <Td dataLabel={columnNames.physicalAdd}>
+                        <Button
+                          variant="link"
+                          isInline
```
Comment (karesti) [inline src/app/ClusterStatus/ClusterStatus.tsx:190]: add 
```xml
 icon={!downloading ? <DownloadIcon/> : null}
```

**Review by karesti — CHANGES_REQUESTED** [review]

**Review by karesti — APPROVED** [review]

**Pull request merged**